src/tests/test_model.py

```diff
@@ -75,7 +75,12 @@ def test_model_vit_fit(datapipe):
     with tempfile.TemporaryDirectory() as tmpdirname:
         # Training
         trainer: L.Trainer = L.Trainer(
-            accelerator="auto",
+            accelerator=(
+                "cpu"  # fallback to CPU on osx-arm64 CI
+                if os.getenv("PYTORCH_MPS_PREFER_METAL") == "0"
+                and torch.backends.mps.is_available()
+                else "auto"
+            ),
```

Comment:
Using this environment variable hack to fallback to CPU on the `macos-14` GitHub Actions runners. If someone with a Mac M1/M2 device can run `python -m pytest --verbose src/tests/` locally and get the unit tests to pass, that would be great!

Reply:
@leothomas, can you do this? My Mac is pre-M1.

Reply:
I'll merge this into the `main` branch first, and then Leo can merge this into the changes in his branch at #166. Might be easier to test this way.